ops/model.py

```diff
@@ -1941,14 +1941,38 @@ def from_name(cls, name: str, message: str):
 
     @classmethod
     def register(cls, child: Type['StatusBase']):
-        """Register a Status for the child's name."""
+        """Class decorator to register a subclass for lookup using :meth:`from_name`.
```

Comment:
This all seems a bit heavy, and actually makes it *more* prominent in the docs. Can we shorten this right up, and add a Sphinx deprecation label, like [`Storage.id`](https://github.com/canonical/operator/blob/20c962360d19f0802c98bbf1878631e4f9122bcb/ops/model.py#L2178), so it'll show up nicely [in the docs](https://ops--1384.org.readthedocs.build/en/1384/#ops.Storage.id)?

I would recommend something more like:

```python
@classmethod
def register(cls, child: Type['StatusBase']):
    """.. deprecated:: 2.17.0 Deprecated - this was for internal use only."""
    warnings.warn('StatusBase.register is for internal use only', DeprecationWarning)
    cls._register(child)
    return child
```